File: source-connector/src/main/java/sourceconnector/repository/file/decompressor/NoneDecompressor.java
package sourceconnector.repository.file.decompressor;

import offsetmanager.domain.file.FileKey;

import java.io.InputStream;

public class NoneDecompressor implements Decompressor {
  @Override
  public boolean supports(FileKey fileKey) {
    return true;
  }

  @Override
  public InputStream decompress(InputStream in) {
    return in;
  }
}

File: source-connector/src/main/kotlin/sourceconnector/repository/file/decompressor/NoneDecompressor.kt
package sourceconnector.repository.file.decompressor

import offsetmanager.domain.file.FileKey
import java.io.InputStream

class NoneDecompressor : Decompressor {
  override fun supports(fileKey: FileKey): Boolean {
    return true
  }

  override fun decompress(`in`: InputStream): InputStream {
    return `in`
  }
}
